### DISCUSSION TITLE
`cnpg` cluster fails due to OOM related to long running transaction

### DISCUSSION THREAD

**Discussion by davinkevin** [cloudnative-pg/cloudnative-pg#5421] [Q&A]
Hello 👋 

As a follow up of this [issue](https://github.com/home-assistant/core/issues/124994), I'm in a situation where my installation, using `cnpg` cluster fails to handle queries done by `home-assistant` during the purge process. 

I've increased resources to the maximum of my local cluster, and even with that the system is not able to "delete" things and finish on time (aka; before reaching the resources.limits.memory and being killed). 

[From the code](https://github.com/home-assistant/core/blob/9082dc2a799734f7dfa7eb7cbcb8e99210796b80/homeassistant/components/recorder/purge.py#L74-L130), there is a very long transaction done by the tool that leads to an Out Of Memory Kill by k8s. 

I'm a bit short on solutions right now. Maintainers reported it as "a db issue". It could be the case, because I expect PG to manage resources and potentially uses disk if there is not enough memory. 

In the other end, the code shouldn't run a so long transaction too. 

tldr; can you help me on the PG part? My [cluster configuration is available](https://gitlab.com/davinkevin.fr/home-server/-/blob/368c5effa16c07d78a3b60a937306c291856a3df/home-assistant/overlays/k8s-server/database/database.yaml) and I'm open to recommendation to manage that case. 

Thank you!
